### PR TITLE
Create OSMigrator

### PR DESCRIPTION
**Pull request by ajvb**
OSMigrator is a tool for openstack migration, with a more flushed out CLI.

### CODE REVIEW

**Comment by jfrost** [comment]
+1 after it passes testing.

File: openstack/migrate-helpers/osmigrator.rb [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  end
+
+  def secgroup_import
+    importer_path = "#{OUTPUT_FOLDER}/#{FILE_NAMES[:secgroups]}"
+    if File.exist?(importer_path)
+      `cd #{OUTPUT_FOLDER} && bash #{FILE_NAMES[:secgroups]}`
+    else
+      puts "ERROR:"
+      puts "Could not find importer script at: #{importer_path}"
+      return
```
Comment (mcasper) [inline openstack/migrate-helpers/osmigrator.rb]: Ruby doesn't need explicit returns unless you're exiting early

File: openstack/migrate-helpers/osmigrator.rb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        export << [node, flavor_image, node_details[0], node_details[1], node_details[2]]
+      end
+    end
+  end
+
+  def flavor_export
+    flavors = `nova flavor-list`.split("\n")[3..-2]
+    # Clean up terminal output
+    flavors = flavors.map{ |f|
+      f = f.split("|").map{ |d|
```
Comment (mcasper) [inline openstack/migrate-helpers/osmigrator.rb]: It's idiomatic in ruby to use `do..end` block syntax if you're going to span multiple lines.

``` Ruby
flavors = flavors.map do |f|
# stuff happens
end
```

Also, the empty parens on the end of methods (like `strip()`) aren't necessary

File: openstack/migrate-helpers/osmigrator.rb [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    CSV.open("#{OUTPUT_FOLDER}/#{FILE_NAMES[:flavors]}", "w",
+            :write_headers => true,
+            :headers => ["name", "id", "ram", "disk", "vcpus", "swap"]) do |export|
+      flavors.each do |flavor|
+        export << [flavor[2], flavor[1], flavor[3], flavor[4], flavor[7], flavor[6]]
+      end
+    end
+  end
+
+  def create_output_folder!
```
Comment (mcasper) [inline openstack/migrate-helpers/osmigrator.rb]: Methods with a bang usually imply that they will raise an exception if they fail, so I think we should either remove the bang here, or not rescue.

**Pull request merged**